ttlcache/cache.go

```diff
@@ -11,6 +11,9 @@ type Cache[T any] interface {
 	Get(ctx context.Context, id string, minimumLifetime time.Duration) (T, bool)
 	// Put adds a value to the cache with the given expiration time.
 	Put(ctx context.Context, id string, value T, expiresAt time.Time)
+	// Delete removes the value with id from the cache. This is typically used
+	// to clean up expired values. It will no-op if the id is not found.
+	Delete(ctx context.Context, id string)
```

Comment:
This is to better handle memory (with the weak cache and with caches that implement it now)